amethyst_network/src/error.rs

```diff
@@ -0,0 +1,85 @@
+//! Module containing error handling logic.
+
+use std::{
+    error::Error as StdError,
+    fmt::{Display, Formatter, Result as FmtResult},
+    io,
+    sync::mpsc,
+};
+
+use crate::server::ServerSocketEvent;
+
+/// The `amethyst_network` result type.
+pub type NetworkError<T> = Result<T, NetworkErrorKind>;
```

Comment:
This should be ` pub type Result<T> = Result<T, Error>;`

We should follow how `std::io::Error` does it. https://doc.rust-lang.org/src/std/io/error.rs.html#66-68

So just create:

```rust
pub struct Error {
  kind: ErrorKind,
}
```

and rename `NetworkErrorKind` to `ErrorKind`, since we have modules we can namespace via that and they can be renamed downstream as needed. This is how std does its errors.